src/i18n/i18nConfig.ts

```diff
@@ -1,10 +1,34 @@
 import i18n from "i18next";
 import LanguageDetector from "i18next-browser-languagedetector";
+import type { CustomDetector } from "i18next-browser-languagedetector";
 import { initReactI18next } from "react-i18next";
+import api from "~/common/lib/api";
 
 import en from "./locales/en.json";
 import hi from "./locales/hi.json";
 
+export const settings: CustomDetector = {
+  name: "settings",
+
+  lookup(_options) {
+    api.getSettings().then((setting) => {
+      i18n.changeLanguage(setting.locale);
+    });
+    return undefined;
```

Comment:
in the [example](https://github.com/i18next/i18next-browser-languageDetector#interface) the language is returned. 
what is this `undefined` doing here? 

I can imagine that when it returns `undefined` it actually says that it did not detect a local and another detector will be used. https://github.com/i18next/i18next-browser-languageDetector/blob/386bc3db9b9e941fa0377840924076530ae60c61/src/index.js#L65

and this PR here might work because it is doing async `i18n.changeLanguage` - but then we don't need this detector class, do we?

Reply:
Me and @escapedcat returned undefined here when we were fixing the TS error, you are right though it makes sense to return the language to match the example. I have updated this to return the language.